hazelcast-client/src/main/java/com/hazelcast/client/impl/statistics/Statistics.java

```diff
@@ -47,7 +48,7 @@
  * statistics to the cluster. If the client statistics feature is enabled,
  * it will be scheduled for periodic statistics collection and sent.
  */
-public class Statistics {
+public class Statistics implements ProbeRegistry.ProbeSource {
```

Comment:
I'm not in favor of having inner interfaces.

Reply:
What belongs together goes together. These interfaces are tightly coupled. None of them makes sense without the other so its easier to get seeing them together.

Reply:
That is what you have packages for.

And Statistics doesn't need to care about ProbeRegistry... But now it does.. it is included in the name it needs to import.

Reply:
Good point. While they are tightly coupled in the complete picture the sources themselves don't have to be involved. This does outweigh benefit of better readability for me. Will extract it.